client/client.go

```diff
@@ -213,10 +208,123 @@ func (client *Client) RemoveToxic(options *ToxicOptions) error {
 
 // ResetState resets the state of all proxies and toxics in Toxiproxy.
 func (client *Client) ResetState() error {
-	resp, err := http.Post(client.endpoint+"/reset", "text/plain", bytes.NewReader([]byte{}))
+	_, err := client.post("/reset", bytes.NewReader([]byte{}))
+	return err
+}
+
+func (c *Client) get(path string) ([]byte, error) {
+	req, err := http.NewRequest("GET", c.endpoint+path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.send(req, http.StatusOK)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	result, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (c *Client) post(path string, body io.Reader) ([]byte, error) {
+	req, err := http.NewRequest("POST", c.endpoint+path, body)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.send(req, http.StatusCreated)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	result, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func (c *Client) patch(path string, body io.Reader) ([]byte, error) {
+	req, err := http.NewRequest("PATCH", c.endpoint+path, body)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.send(req, http.StatusOK)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	result, err := io.ReadAll(resp.Body)
```

Comment:
I wonder if this should be part of `send` instead, since it's duplicated in 3 other function

Reply:
good catch.